### PR TITLE
renamed dictionary variable addresses #27166

### PR DESCRIPTION
**Pull request by jubebo**
As dicussed in [issue 27166](https://github.com/apache/beam/issues/27166), broke down the changes of [PR 27166](https://github.com/apache/beam/pull/27199) and started with renaming of `dict_of_tuples` to `field_names_and_types`.

The entries in `field_names_and_types` could have the following content:
```python
print(field_names_and_types)
>>> [('string_feature_nullable', typing.Union[str, NoneType]),
>>>  ('timestamp_feature_nullable', typing.Union[apache_beam.utils.timestamp.Timestamp, NoneType]),
>>>  ('float_feature_repeated', typing.Sequence[numpy.float64]),
>>>  ('bytes_features_required', <class 'bytes'>]
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py [hunk 1]
```diff
@@ -76,9 +76,9 @@ def generate_user_type_from_bq_schema(the_table_schema):
       raise ValueError(
           f"Encountered "
           f"an unsupported type: {the_schema['fields'][i]['type']!r}")
-    # TODO svetaksundhar@: Map remaining BQ types
-    dict_of_tuples.append((the_schema['fields'][i]['name'], typ))
-  sample_schema = beam.typehints.schemas.named_fields_to_schema(dict_of_tuples)
+    # TODO @svetaksundhar: Map remaining BQ types
```
Comment (svetakvsundhar) [inline sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py:79]: I think we can delete this TODO as it's already specified [here](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py#LL51C5-L52C39)

**Review by svetakvsundhar — CHANGES_REQUESTED** [review]

**Comment by svetakvsundhar** [comment]
Run Python 3.7 PostCommit

File: sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py (hunk 1 above)
Comment (jubebo) [reply sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py]: Deleted this line as suggested

**Review by svetakvsundhar — APPROVED** [review]
Thanks! In the PR description, can we print an example of `field_names_and_types`, to make it easier for the other reviewers to understand why this change is needed?

**Comment by svetakvsundhar** [comment]
R: @tvalentyn 

cc: @TheNeuralBit

**Comment by jubebo** [comment]
Added example of `field_names_and_types` as suggested.

**Comment by tvalentyn** [comment]
Looks good, we can merge when tests pass.

**Pull request merged**